### ISSUE TITLE
Sparse network weights

### ISSUE THREAD

**Issue by mjyeh** [lightvector/KataGo#210]
Hi! Out of curiosity, if a term encouraging sparsity in the network weights (for example, group lasso) were added to the loss function for training, do you think this would reduce the time needed for a forward pass through the network and allow for more self-play games? Also, do you think this would greatly impact network strength? Thanks!

**Comment by lightvector** [comment]
I have no idea. But you don't get real answers to questions like these without a lot of both implementation work and empirical research work, neither of which I'm in a good position to do in terms of my expertise. There are still like a dozen other major untested ideas and papers and things that might help too, the bottleneck is always the actual implementation and research.  Given that you said "Out of curiosity", I don't suppose you'd be interested in being the one to do the work?  :)

Find the most promising and widely-cited papers about each part of this (most papers seem to focus primarily on compressability and space or on compute benefits for CPUs or embedded systems, rather than compute benefits for GPUs), find some actual libraries that do sparse convolution efficiently on the GPU, integrate them into an implementation, and actually train a neural network and test it out.

**Comment by mjyeh** [comment]
Fair enough. Looks like I've got some work to do then. :)
Thank you for the response!
